rest/route/user_test.go

```diff
@@ -1011,3 +1011,89 @@ func TestOffboardUserHandlerAdminis(t *testing.T) {
+		PatchNumber:  12,
+		OnlyAPI:      true,
+		SystemRoles: []string{
+			"bot_access",
+		},
+	}
+	me := user.DBUser{
+		Id:           "me",
+		EmailAddress: "i@rock.com",
+		APIKey:       "my_key",
+	}
+	assert.NoError(t, usrToRetrieve.Insert())
+	assert.NoError(t, me.Insert())
+
+	for testName, testCase := range map[string]func(t *testing.T){
+		"UserNotFound": func(t *testing.T) {
+			req, err := http.NewRequest(http.MethodGet, "http://example.com/api/rest/v2/users/no_one", nil)
+			req = gimlet.SetURLVars(req, map[string]string{"user_id": "no_one"})
+
+			require.NoError(t, err)
+			handler := makeGetUserHandler()
+
+			assert.NoError(t, handler.Parse(ctx, req))
+			assert.Equal(t, handler.(*getUserHandler).userId, "no_one")
+
+			resp := handler.Run(gimlet.AttachUser(context.Background(), &me))
+			assert.Equal(t, resp.Status(), http.StatusNotFound)
+		}, "UserFound": func(t *testing.T) {
+			req, err := http.NewRequest(http.MethodGet, "http://example.com/api/rest/v2/users/beep.boop", nil)
+			req = gimlet.SetURLVars(req, map[string]string{"user_id": "beep.boop"})
+
+			require.NoError(t, err)
+			handler := makeGetUserHandler()
+
+			assert.NoError(t, handler.Parse(ctx, req))
+			assert.Equal(t, handler.(*getUserHandler).userId, "beep.boop")
+
+			resp := handler.Run(gimlet.AttachUser(context.Background(), &me))
+			assert.Equal(t, resp.Status(), http.StatusOK)
+			respUsr, ok := resp.Data().(*restModel.APIDBUser)
+			require.True(t, ok)
+			assert.NotEmpty(t, respUsr)
+			assert.Equal(t, usrToRetrieve.Id, utility.FromStringPtr(respUsr.UserID))
+			assert.Equal(t, usrToRetrieve.DisplayName(), utility.FromStringPtr(respUsr.DisplayName))
+			assert.Equal(t, usrToRetrieve.EmailAddress, utility.FromStringPtr(respUsr.EmailAddress))
+			assert.Equal(t, usrToRetrieve.OnlyAPI, respUsr.OnlyApi)
+			assert.Equal(t, usrToRetrieve.Roles(), respUsr.Roles)
+		}, "DisplayNameUser": func(t *testing.T) {
+			req, err := http.NewRequest(http.MethodGet, "http://example.com/api/rest/v2/users/robots are good", nil)
+			req = gimlet.SetURLVars(req, map[string]string{"user_id": usrToRetrieve.DispName})
+
+			require.NoError(t, err)
+			handler := makeGetUserHandler()
+
+			assert.NoError(t, handler.Parse(ctx, req))
+			assert.Equal(t, handler.(*getUserHandler).userId, "robots are good")
+
+			resp := handler.Run(gimlet.AttachUser(context.Background(), &me))
+			assert.Equal(t, resp.Status(), http.StatusOK)
+			respUsr, ok := resp.Data().(*restModel.APIDBUser)
+			require.True(t, ok)
+			assert.NotEmpty(t, respUsr)
+			assert.Equal(t, usrToRetrieve.Id, utility.FromStringPtr(respUsr.UserID))
+			assert.Equal(t, usrToRetrieve.DisplayName(), utility.FromStringPtr(respUsr.DisplayName))
+			assert.Equal(t, usrToRetrieve.EmailAddress, utility.FromStringPtr(respUsr.EmailAddress))
+			assert.Equal(t, usrToRetrieve.OnlyAPI, respUsr.OnlyApi)
+			assert.Equal(t, usrToRetrieve.Roles(), respUsr.Roles)
+		},
+	} {
+		t.Run(testName, func(t *testing.T) {
```

Comment:
This seems a little weird way to write the test cases because this is pretty much the same as just listing out `t.Run` several times. Usually when we put test cases in a loop, it's so that they can do common setup/cleanup in the `t.Run` body, and then run without worrying about whether a previous test has modified the original setup. Can we either move the setup (L1016-1037) into this loop body?

Reply:
I don't want to move the setup in because then the tests can't reference those fields without passing them around, which seems unnecessary given that the values won't change. I can just move these out of the loop completely and write two different tests if you prefer, however I like using the map structure for organization.

Reply:
Actually I can move the inserts in, which I think still fits the bill.